### PR TITLE
IREE build is made optional, fix models to allow onnx export, handle multiple input onnx export

### PR DESCRIPTION
**Pull request by kumardeepakamd**

### CODE REVIEW

File: e2eshark/run.py [hunk 1]
```diff
@@ -1157,9 +1148,13 @@ def main():
         help="Please select a dir with large free space to cache all torch, hf, turbine_tank model data",
         required=True,
     )
-    
+
     args = parser.parse_args()
     cache_dir = args.cachedir
+    if not os.path.exists(cache_dir):
```
Comment (saienduri) [inline e2eshark/run.py:1154]: I have this in other PR. Maybe remove this so we have everything related to cache_dir documented there?

**Review by saienduri — APPROVED** [review]
Looks good to me! Nice that user has option to skip the whole iree build now.

File: e2eshark/run.py [hunk 2]
```diff
@@ -333,8 +324,12 @@ def runCodeGeneration(
         print(f"Test {testName} failed [{curphase}]")
         return 1
     logfilename = curphase + ".log"
-    commandname = (
-        "/tools/iree-compile --iree-input-demote-i64-to-i32 --iree-hal-target-backends="
+    commandname = ""
+    if SHARED_IREE_BUILD:
+        commandname = SHARED_IREE_BUILD + "/tools/"
```
Comment (ScottTodd) [inline e2eshark/run.py:329]: Binary (non-Python) distributions are also useful to support. Those don't include a `tools` folder (they are flatter)

e.g.
```
iree-dist/
  bin/
    iree-compile
    iree-run-module
  include/
    iree/...
      (header files)
  lib/
    libIREECompiler.so
```

The most generic way to support both is to just point at the binary directory, either `bin/` from an install/dist like that or `tools/` from a build directory

**Review by ScottTodd — COMMENTED** [review]
Is https://github.com/nod-ai/SHARK-TestSuite/issues/49 fixed now?

**Pull request merged**